source/store/actions/dynamicFormData.ts

```diff
@@ -0,0 +1,106 @@
+import { Form } from '../../types/FormTypes';
+import { Case } from '../../types/CaseType';
+import { User } from '../../types/UserTypes';
+
+const addIds = (descriptor: string[], questionId: string, formId: string) =>
+  descriptor.map(s => {
+    if (s.split('.')[0] !== 'user' && questionId !== '') {
+      return `${formId}.${questionId}.${s}`;
+    }
+    if (s.split('.')[0] !== 'user' && questionId === '') {
+      return `${formId}.${s}`;
+    }
+    return s;
+  });
+
+/** Takes a form and generates the dataMap template used to fill the dynamical data.  */
+const makeDataMap = (form: Form) => {
+  const dataMap = {};
+  form.steps.forEach(step => {
+    step.questions.forEach(qs => {
+      if (qs.loadPrevious) {
+        dataMap[qs.id] = addIds(qs.loadPrevious, '', form.id);
+      } else if (qs.type === 'substepList' && qs.items) {
+        const items = {};
+        qs.items.forEach(item => {
+          if (item.loadPrevious) {
+            items[item.title] = addIds(item.loadPrevious, qs.id, form.id);
+          }
+        });
+        if (Object.keys(items).length > 0) {
+          dataMap[qs.id] = items;
+        }
+      } else if (qs.type === 'editableList' && qs.inputs) {
+        const inputs = {};
+        qs.inputs.forEach(input => {
+          if (input.loadPrevious) {
+            inputs[input.key] = addIds(input.loadPrevious, qs.id, form.id);
+          }
+        });
+        if (Object.keys(inputs).length > 0) {
+          dataMap[qs.id] = inputs;
+        }
+      }
+    });
+  });
+  return dataMap;
+};
+
+const sortCasesByLastUpdated = (list: Case[]) => list.sort((a, b) => b.updatedAt - a.updatedAt);
+
+const treeParseAcc = (
```

Comment:
Hard to know what this function does by looking at the function name. What's the difference between this and treeParser and what does Acc stand for.